library/choose_disk.py

```diff
@@ -0,0 +1,436 @@
+            match_count = 0
+            # Checking what features are matching
+            for feature in current_lookup:
+                if feature not in current_physical_disk:
+                    continue
+
+                # Default operator is equal
+                operator = "_equal"
+
+                # Assign left and right operands
+                right = current_lookup[feature]
+                left = current_physical_disk[feature]
+
+                # Test if we have anoter operator
+                arguments = _REGEXP.search(right)
+                if arguments:
+                        new_operator = "_" + arguments.group(1)
+                        # Check if the associated function exists
+                        if new_operator in globals():
+                            # and assign operands with the new values
+                            operator = new_operator
+                            right = arguments.group(2)
+                        else:
+                            fatal("Unsupported %s operator in : %s" % (new_operator, right))
+
+                # Let's check if (left <operator> right) is True meaning the match is done
+                if globals()[operator](convert_units(left), convert_units(right)):
+                    debug("  %s : match  %s %s %s" % (physical_disk, left, operator, right))
+                    match_count = match_count + 1
+                    continue
+                else:
+                    debug("  %s : no match  %s %s %s" % (physical_disk, left, operator, right))
+                    match_count = match_count
+                    # nomatch
+
+            # If all the features matched
+            if match_count == len(current_lookup):
+                info("  %50s matched" % (physical_disk))
+                # When looking for an infinite number of disks, we can have
+                # several disks per matching
+                if disk not in matched_devices:
+                    matched_devices[disk] = []
+                matched_devices[disk].append(physical_disks[physical_disk])
+                exclude_list.append(physical_disk)
+                # If we look for an inifinite list of those devices, let's
+                # continue looking for the same description unless let's go to
+                # the next device
+                if infinite is False:
+                    break
+            # We were unable to find all part of the required features
+            elif match_count > 0:
+                info("  %50s partially matched with %d/%d items" % (physical_disk, match_count, len(current_lookup)))
+            else:
+                info("  %50s no devices matched" % (physical_disk))
+
+    final_list = {}
+    for matched_device in matched_devices.keys():
+        for n in range(0, len(matched_devices[matched_device]), 1):
+            name = matched_device
+            if len(matched_devices[matched_device]) > 1:
+                name = "%s_%03d" % (matched_device, n)
```

Comment:
I am not sure what this naming convention (`matched_device_n`) is for? Is this an internal bookkeeping?